Assets/Project/Modules/VFX/Generic/Scripts/ParticleFactory/ParticleFactory.cs

```diff
@@ -17,9 +17,17 @@ public ParticleFactory(ParticleFactoryConfig config, Transform parent)
             _typeToPrefab = _config.GetTypeToPoolDictionary(parent);
         }
 
-        public Transform Create(ParticleTypes type, Vector3 position, Quaternion rotation)
+        public Transform Create(ParticleTypes type, Vector3 position, Quaternion rotation, Transform parent = null)
```

Comment:
If the spawned gameObjects are all InterpolatorRecycleParticle, then make it return that class (Do the GetComponent) instead of returning Transform